### PR TITLE
feat(api): added reports

### PR DESCRIPTION
**Pull request by BronzW**

### CODE REVIEW

**Review by heybereket — APPROVED** [review]

**Comment by HenryLeC** [comment]
I will review in a min

File: packages/api/schema.gql [hunk 1]
```diff
@@ -255,6 +263,20 @@ type Repo {
...
 }
 
+type Report {
+  accuse: User!
+  createdAt: String!
+  id: ID!
+  information: String
+  reporter: User!
+  resolved: Boolean!
+  type: String!
```
Comment (HenryLeC) [inline packages/api/schema.gql]: Make this a report type

File: packages/api/src/modules/reports/GetQueue.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import Report from '@entities/Report';
+import { Role } from '@modules/user/Roles';
+import { ContextType } from '@root/apolloServer';
+import { ApolloError } from 'apollo-server-express';
+import { Arg, Authorized, Ctx, Query, Resolver, Root } from 'type-graphql';
+
+@Resolver()
+export class GetQueueResolver {
+  @Authorized()
```
Comment (HenryLeC) [inline packages/api/src/modules/reports/GetQueue.resolver.ts]: Can you make this authorized with either admin or mod role

File: packages/api/src/modules/reports/GetQueue.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  async getQueue(@Ctx() { getUser }: ContextType) {
+    const user = await getUser();
+
+    // if (!user.roles.includes(Role.Moderator)) {
+    //   throw new ApolloError("You don't have permission to do that!");
+    // }
+
+    return await Report.find({
+      where: { resolved: false },
+      relations: ['accuse', 'reporter'],
```
Comment (HenryLeC) [inline packages/api/src/modules/reports/GetQueue.resolver.ts]: This isn't needed if they are promises

File: packages/api/src/modules/reports/MarkAsResolved.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import Report from '@entities/Report';
+import { Role } from '@modules/user/Roles';
+import { ContextType } from '@root/apolloServer';
+import { ApolloError } from 'apollo-server-express';
+import { Arg, Authorized, Ctx, Mutation, Resolver } from 'type-graphql';
+
+@Resolver()
+export class MarkAsResolvedResolver {
+  @Authorized()
```
Comment (HenryLeC) [inline packages/api/src/modules/reports/MarkAsResolved.resolver.ts]: Again mod or admin

**Review by HenryLeC — CHANGES_REQUESTED** [review]
Just a few changes

**Comment by HenryLeC** [comment]
I really don't like merging when I can't test so I will test it when I get home and then merge

**Comment by BronzW** [comment]
> I really don't like merging when I can't test so I will test it when I get home and then merge

do so

**Review by HenryLeC — APPROVED** [review]

**Pull request merged**